### PR TITLE
add - identifier in api

### PR DESCRIPTION
**Pull request by khemrajrathore**

### CODE REVIEW

**Comment by pandurangpatil** [comment]
@HiteshMah-Jan can you go through this PR and approve the same?

File: config/systemConfig/java.yaml [hunk 1]
```diff
@@ -7,3 +7,6 @@ systemConfig:
   
   - key: apiSinks
     value: (?i)(?:url|client|openConnection|request|execute|newCall|load|host|access|fetch|get|getInputStream|getApod|getForObject|getForEntity|list|set|put|post|proceed|trace|patch|Path|send|sendAsync|remove|delete|write|read|assignment|provider|exchange|postForEntity)
+
+  - key: apiIdentifier
+    value: (?i).*((hook|base|authorize|provider|endpoint|installation)(s){0,1}_url|(slack|web)_hook|(rest|api)_endpoint).*
```
Comment (HiteshMah-Jan) [inline config/systemConfig/java.yaml:12]: Any reason to have all these regex not match CamelCase

File: config/systemConfig/python.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+systemConfig:
+  - key: apiHttpLibraries
+    value: (?i)(request|aiohttp|treq|grequests|urllib|http|uplink|httoop|flask_restful|tornado.httpclient|pycurl|bs4|.*(HttpClient)).*
```
Comment (HiteshMah-Jan) [inline config/systemConfig/python.yaml:3]: Not sure if we can make api calls using `bs4`. can we get this checked ?

File: rules/sinks/third_parties/api/java.yaml [hunk 3]
```diff
@@ -4,4 +4,5 @@ sinks:
     name: Third Party API
     patterns:   
       - "(?i)((?:http|https):\\/\\/[a-zA-Z0-9_-][^)\\/(#|,!>\\s]{1,50}\\.(?:com|net|org|de|in|uk|us|io|gov|cn|ml|ai|ly|dev|cloud|me|icu|ru|info|top|tk|tr|cn|ga|cf|nl)).*(?<!png|jpeg|jpg|txt|blob|css|html|js|svg)"
+      - "(?i).*((hook|base|authorize|provider|endpoint|installation)(s){0,1}_url|(slack|web)_hook|(rest|api)_endpoint).*"
```
Comment (HiteshMah-Jan) [inline rules/sinks/third_parties/api/java.yaml:7]: We need support for these to be matched with identifiers or members. can you confirm if it is not just literals ?

**Review by HiteshMah-Jan — DISMISSED** [review]

File: config/systemConfig/python.yaml (hunk 2 above)
Comment (khemrajrathore) [reply config/systemConfig/python.yaml:3]: @dbMundada @hiteshbedre this is the python api libraries, have moved them to privado from privado-core

File: rules/sinks/third_parties/api/java.yaml (hunk 3 above)
Comment (khemrajrathore) [reply rules/sinks/third_parties/api/java.yaml:7]: this will only be matched with literals

**Comment by pandurangpatil** [comment]
@khemrajrathore can we get the open questions resolved, so that we can merge this request.

File: config/systemConfig/python.yaml (hunk 2 above)
Comment (khemrajrathore) [reply config/systemConfig/python.yaml:3]: It is a web scrapping library from which XML or HTML content can be obtained. Let's keep it for now

File: config/systemConfig/java.yaml (hunk 1 above)
Comment (khemrajrathore) [reply config/systemConfig/java.yaml:12]: Let me update the regex to have `_` as optional

File: config/systemConfig/python.yaml (hunk 2 above)
Comment (dbMundada) [reply config/systemConfig/python.yaml:3]: Right, @khemrajrathore  bs4 used to scrape the websites & get the data.

**Review by pandurangpatil — APPROVED** [review]
Looks good

**Pull request merged**